cpp/src/parquet/page_index.cc

```diff
@@ -533,6 +569,21 @@ class ColumnIndexBuilderImpl final : public ColumnIndexBuilder {
     /// Decide the boundary order from decoded min/max values.
     auto boundary_order = DetermineBoundaryOrder(min_values, max_values);
     column_index_.__set_boundary_order(ToThrift(boundary_order));
+
+    /// Finalize level histogram.
+    const int64_t num_pages = column_index_.null_pages.size();
+    const int64_t rep_level_hist_size = column_index_.repetition_level_histograms.size();
+    const int64_t def_level_hist_size = column_index_.definition_level_histograms.size();
+    if (rep_level_hist_size == (descr_->max_repetition_level() + 1) * num_pages) {
+      column_index_.__isset.repetition_level_histograms = true;
+    } else {
```

Comment:
Hmm, are we going to silently clear the histogram if the length is not right? Shouldn't we error out instead?

Reply:
Make sense. I was intended to not let it fail. But it would be a writer bug if it happens.